Bug? quasar upgrade -i still keep going Yarn not PNPM (default)

### What happened?

`PS D:\mycode\quasar\vue> quasar upgrade -i

    Quasar CLI · Gathering information with yarn...

 @quasar/extras: 1.15.6 → 1.15.7

  Quasar CLI · Running "yarn add @quasar/extras@^1.15.7"

yarn add v1.22.19
info No lockfile found.
[1/5] Validating package.json...
[2/5] Resolving packages...
warning vue > @vue/compiler-sfc > magic-string > sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
warning @quasar/app-webpack > cssnano > cssnano-preset-default > postcss-svgo > svgo > stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
warning unplugin-auto-import > magic-string > sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
[3/5] Fetching packages...
[4/5] Linking dependencies...
error An unexpected error occurred: "ENOENT: no such file or directory, copyfile ....`

### What did you expect to happen?

env: npm,yarn,pnpm all installed in my system for diff old project

and new project:
pnpm create quasar  => set pnpm by default

quasar upgrade -i should use pnpm not yarn
this will mess up package stores

### Reproduction URL

nop

### How to reproduce?

pnpm create quasar  
=> set pnpm by default
=> quasar upgrade -i

### Flavour

Quasar CLI with Webpack (@quasar/cli | @quasar/app-webpack)

### Areas

Quasar CLI Commands/Configuration (@quasar/cli | @quasar/app-webpack | @quasar/app-vite)

### Platforms/Browsers

Chrome

### Quasar info output

```Shell
Quasar CLI · Gathering information with yarn...

 @quasar/extras: 1.15.6 → 1.15.7

  Quasar CLI · Running "yarn add @quasar/extras@^1.15.7"

yarn add v1.22.19
info No lockfile found.
[1/5] Validating package.json...
[2/5] Resolving packages...
warning vue > @vue/compiler-sfc > magic-string > sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
warning @quasar/app-webpack > cssnano > cssnano-preset-default > postcss-svgo > svgo > stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
warning unplugin-auto-import > magic-string > sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
[3/5] Fetching packages...
[4/5] Linking dependencies...
error An unexpected error occurred: "ENOENT: no such file or directory, copyfile
```


### Relevant log output

_No response_

### Additional context

_No response_

if i 
`pnpm add @quasar/extras@^1.15.7`

then i lost
Error: Cannot find module 'webpack'


` Dev mode.................. spa
 Pkg quasar................ v2.10.2
 Pkg @quasar/app-webpack... v3.6.1
 Pkg webpack............... v5
 Debugging................. enabled

 Configured browser support (>= 89.18% of global marketshare):
 · Chrome for Android >= 107
 · Firefox for Android >= 106
 · Android >= 107
 · Chrome >= 99
 · Edge >= 104
 · Firefox >= 98
 · iOS >= 13.4-13.7
 · Opera >= 88
 · Safari >= 15.1

[  Error: Cannot find module 'webpack'
  Require stack:
  - D:\mycode\quasar_oms\oms\quasar.config.js
  - D:\mycode\quasar_oms\oms\node_modules\.pnpm\@quasar+app-webpack@3.6.1_ggkzqseu57njbmlcsujprzzlla\node_modules\@quasar\app-webpack\lib\quasar-conf-file.js 
  - D:\mycode\quasar_oms\oms\node_modules\.pnpm\@quasar+app-webpack@3.6.1_ggkzqseu57njbmlcsujprzzlla\node_modules\@quasar\app-webpack\lib\cmd\dev.js
  - D:\mycode\quasar_oms\oms\node_modules\.pnpm\@quasar+app-webpack@3.6.1_ggkzqseu57njbmlcsujprzzlla\node_modules\@quasar\app-webpack\bin\quasar
  - D:\nodejs\cache_folder\node_global\node_modules\@quasar\cli\bin\quasar

  - loader:933 Function.Module._resolveFilename
    node:internal/modules/cjs/loader:933:15

  - loader:778 Function.Module._load
    node:internal/modules/cjs/loader:778:27

  - loader:1005 Module.require
    node:internal/modules/cjs/loader:1005:19

    node:internal/modules/cjs/loader:1155:10

  - loader:981 Module.load
    node:internal/modules/cjs/loader:981:32

  - loader:822 Function.Module._load
    node:internal/modules/cjs/loader:822:12

  - loader:1005 Module.require
    node:internal/modules/cjs/loader:1005:19

] {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\mycode\\quasar_oms\\oms\\quasar.config.js',
    'D:\\mycode\\quasar_oms\\oms\\node_modules\\.pnpm\\@quasar+app-webpack@3.6.1_ggkzqseu57njbmlcsujprzzlla\\node_modules\\@quasar\\app-webpack\\lib\\quasar-conf-file.js',
    'D:\\mycode\\quasar_oms\\oms\\node_modules\\.pnpm\\@quasar+app-webpack@3.6.1_ggkzqseu57njbmlcsujprzzlla\\node_modules\\@quasar\\app-webpack\\lib\\cmd\\dev.js',
    'D:\\mycode\\quasar_oms\\oms\\node_modules\\.pnpm\\@quasar+app-webpack@3.6.1_ggkzqseu57njbmlcsujprzzlla\\node_modules\\@quasar\\app-webpack\\bin\\quasar',    'D:\\nodejs\\cache_folder\\node_global\\node_modules\\@quasar\\cli\\bin\\quasar'
  ]
}`

--- a/cli/lib/node-packager.js
+++ b/cli/lib/node-packager.js
@@ -14,8 +14,14 @@ function isInstalled (cmd) {
 }
 
 function getPackager (root) {
+
+
   if (fs.existsSync(join(root, 'yarn.lock'))) {
     return 'yarn'
+  }
+
+  if (fs.existsSync(join(root, 'pnpm-lock.yaml'))) {
+    return 'pnpm'
   }
 
   if (fs.existsSync(join(root, 'package-lock.json'))) {
@@ -26,11 +32,15 @@ function getPackager (root) {
     return 'yarn'
   }
 
+  if (isInstalled('pnpm')) {
+    return 'pnpm'
+  }
+
   if (isInstalled('npm')) {
     return 'npm'
   }
 
-  fatal('⚠️  Please install Yarn or NPM before running this command.\n')
+  fatal('⚠️  Please install Yarn or Pnpm or NPM before running this command.\n')
 }
 
 module.exports = getPackager

--- a/cli/lib/node-packager.js
+++ b/cli/lib/node-packager.js
@@ -14,8 +14,6 @@ function isInstalled (cmd) {
 }
 
 function getPackager (root) {
-
-
   if (fs.existsSync(join(root, 'yarn.lock'))) {
     return 'yarn'
   }
@@ -32,12 +30,12 @@ function getPackager (root) {
     return 'yarn'
   }
 
-  if (isInstalled('pnpm')) {
-    return 'pnpm'
-  }
-
   if (isInstalled('npm')) {
     return 'npm'
+  }
+
+  if (isInstalled('pnpm')) {
+    return 'pnpm'
   }
 
   fatal('⚠️  Please install Yarn or Pnpm or NPM before running this command.\n')

